script.js

```diff
@@ -79,48 +78,102 @@ const projects = [{
+        if (i > 0) {
+          const str = '';
+          det.innerText = str + projects[projectNumber - 1].details[i];
+          det.classList.add('grey-color');
+        } else {
+          det.innerText = projects[projectNumber - 1].details[i];
+        }
+        det.classList.add('item');
+        popupDetails.appendChild(det);
+      }
+      liveUrl.href = projects[projectNumber - 1].linkLive;
+      sourceUrl.href = projects[projectNumber - 1].linkSource;
     });
-    popup.insertBefore(imageTag, popupDescription);
+  });
+  closePopup.addEventListener('click', () => {
+    popupContainer.style.display = 'none';
+  });
+}
 
-    popupContainer.style.display = 'flex';
-    for (let i = 0; i < projects[projectNumber - 1].details.length; i += 1) {
-      const det = document.createElement('span');
-      if (i > 0) {
-        const str = '';
-        det.innerText = str + projects[projectNumber - 1].details[i];
-        det.classList.add('grey-color');
-      } else {
-        det.innerText = projects[projectNumber - 1].details[i];
-      }
-      det.classList.add('item');
-      popupDetails.appendChild(det);
+// Form validation
+const contactForm = document.getElementById('contact-form');
+const email = document.getElementById('email');
+const errorMessage = document.querySelector('.error-message');
+
+function emailValidation() {
+  const re = /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
+  return re.test(String(email.value));
+}
+
+function emailLowerCheck() {
+  const arr = email.value.split('');
+  let result = true;
+  arr.forEach((x) => {
+    if (
+      !(
+        (x.charCodeAt() >= 97 && x.charCodeAt() <= 122)
+        || x.charCodeAt() === 64
+        || x.charCodeAt() === 46
+      )
+    ) {
+      result = false;
     }
-    liveUrl.href = projects[projectNumber - 1].linkLive;
-    sourceUrl.href = projects[projectNumber - 1].linkSource;
   });
-});
+  return result;
+}
+
+function formValidation() {
+  contactForm.addEventListener('submit', (e) => {
+    e.preventDefault();
+    if (email.value.trim() === '') {
+      errorMessage.style.display = 'block';
+      errorMessage.innerText = 'Email is required!';
+    } else if (emailValidation() === false) {
+      errorMessage.style.display = 'block';
+      errorMessage.innerText = 'This is not a valid email.';
+    } else if (emailLowerCheck() === false) {
+      errorMessage.style.display = 'block';
+      errorMessage.innerText = 'Please write your email in lower case.';
+      errorMessage.scrollIntoView(true);
+    } else {
+      errorMessage.style.display = 'none';
+      contactForm.submit();
+    }
+  });
+}
+
+function main() {
+  openCloseMenu();
+  popUpProject();
+  formValidation();
+}
 
-closePopup.addEventListener('click', () => {
-  popupContainer.style.display = 'none';
-});
-// End popup window
+main();
```

Comment:
[OPTIONAL] I think it's better if you call the `main` function in the `window.onload` event. Just like this `window.onload = main;`. This way it will be executed after the window finishes loading.

I hope this is clarifying.